File: ESAMI/DOCS_NOTES/Fondamenti di Informatica II/AlgoritmiEStruttureDati-master/AlgoritmiEStruttureDati-master/Algoritmi - Esami/ALGORITMI/giugno2017/java-aux/BST.java
public class BST{

    private BinNode root;

    public BST() {
        this.root = null;
    }

    public BinNode BST_insert(int k) {
        /*DA IMPLEMENTARE*/
    	return null; //questa istruzione è stata inserita per permettere la compilazione e va rimossa
    }

    public BinNode BST_find(int k) {
        /*DA IMPLEMENTARE*/
    	return null; //questa istruzione è stata inserita per permettere la compilazione e va rimossa
    }

    int maxOnLevel (int level){
        /*DA IMPLEMENTARE*/
    	return 0; //questa istruzione è stata inserita per permettere la compilazione e va rimossa
    }

    private void print(BinNode t, int level) {

        if (t == null)
            return;

        for (int i = 0; i < level - 1; i++) {
            System.out.print("   ");
        }

        if (level > 0) {
            System.out.print(" |--");
        }

        System.out.println(t.getKey());

        print(t.getLeft(), level + 1);
        print(t.getRight(), level + 1);
    }

    void BST_print(){
        print(this.root, 0);
    }

    

}

File: ESAMI/DOCS_NOTES/Fondamenti di Informatica II/AlgoritmiEStruttureDati-master/AlgoritmiEStruttureDati-master/Algoritmi - Esami/ALGORITMI/Esami/Esami/2017/2Nov/c-aux/bst.c
#include <stdlib.h>
#include <stdio.h>
#include "bst.h"
#include "queue.h"


BinNode * newBinNode(int k) {
    /*OK*/
    BinNode * b = malloc(sizeof(BinNode));
	b->key = k;
	b->left = NULL;
	b->right = NULL;

	return b;
}

BST * newBST() {
    /*OK*/
    BST * b = malloc(sizeof(BST));
    b->root = NULL;

	return b;
}

// altri prototipi (per funzioni locali)
static void print(BinNode *t, int level, char pos);
// fine prototipi



static void print(BinNode *t, int level, char pos) {
    if(t == NULL) return;
    for (int i = 0; i < level - 1; i++) printf("   ");
    if (level > 0) printf(" %c:--", pos);

    printf("%d\n", t->key);
    print(t->left, level + 1, 'l');
    print(t->right, level + 1, 'r');
}

void BST_print(BST *t){
    if(t != NULL && t->root != NULL) print(t->root, 0, ' ');
}

BinNode *BST_insert(BST *t, int k) {
    /* DA IMPLEMENTARE */
    return NULL;
}

int isBST(BST *t) {
    /* DA IMPLEMENTARE */
    return -1;
}


int isBalanced(BST *t) {
    /* DA IMPLEMENTARE */
    return -1;
}


int isAVL(BST *t) {
    /* DA IMPLEMENTARE */
    return -1;
}